notifier/util_bench_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package notifier
+
+import (
+	"testing"
+	"time"
+
+	"github.com/prometheus/prometheus/model/labels"
+)
+
+// createAlerts generates a slice of test alerts with the specified count.
+func createAlerts(alertCount, labelCount, annotationCount int) []*Alert {
```

Comment:
thanks for this.
note that the benchmarks can be part of `util_test.go`.

could you elaborate on the added cases? I see they only exercise `alertsToOpenAPIAlerts` which is same on main and in this PR. A benchmark that would help us quantify the overhead would cover more parts; the one that changed.

if you think `alertsToOpenAPIAlerts` itself can be optimized, let's keep that for another PR.

Reply:
I added the benchmark following this comment https://github.com/prometheus/prometheus/pull/16355#discussion_r2314586060